syntax = "proto3";

option go_package = "services/proto";

message UpdateDetails {
  int64 user_id = 1;
  int64 article_id = 2;
  string body = 3;
  string title = 4;
  repeated string tags = 5;
  string summary = 6;
}

message UpdateResponse {
  enum ResultType {
    OK = 0;
    ERROR = 1;
  }

  ResultType result_type = 1;

  // Should only be set if result_type is not OK
  string error = 2;
}

message ReceivedUpdateDetails {
}


// Service for sending and receiving server-to-server update activities.
service S2SUpdate {
  rpc ReceiveUpdateActivity(ReceivedUpdateDetails) returns (UpdateResponse);
  rpc SendUpdateActivity(UpdateDetails) returns (UpdateResponse);
}
